File: src/Cxx/Meshes/BoundaryEdges.cxx
#include <vtkSmartPointer.h>
#include <vtkFeatureEdges.h>
#include <vtkPolyData.h>
#include <vtkDiskSource.h>
#include <vtkPolyDataMapper.h>
#include <vtkActor.h>
#include <vtkRenderWindow.h>
#include <vtkRenderer.h>
#include <vtkRenderWindowInteractor.h>

int main(int, char *[])
{
  vtkSmartPointer<vtkDiskSource> diskSource =
    vtkSmartPointer<vtkDiskSource>::New();
  diskSource->Update();

  vtkSmartPointer<vtkFeatureEdges> featureEdges =
    vtkSmartPointer<vtkFeatureEdges>::New();
  featureEdges->SetInputConnection(diskSource->GetOutputPort());
  featureEdges->BoundaryEdgesOn();
  featureEdges->FeatureEdgesOff();
  featureEdges->ManifoldEdgesOff();
  featureEdges->NonManifoldEdgesOff();
  featureEdges->Update();

  // Visualize
  vtkSmartPointer<vtkPolyDataMapper> edgeMapper =
    vtkSmartPointer<vtkPolyDataMapper>::New();
  edgeMapper->SetInputConnection(featureEdges->GetOutputPort());
  vtkSmartPointer<vtkActor> edgeActor =
    vtkSmartPointer<vtkActor>::New();
  edgeActor->SetMapper(edgeMapper);

  vtkSmartPointer<vtkPolyDataMapper> diskMapper =
    vtkSmartPointer<vtkPolyDataMapper>::New();
  diskMapper->SetInputConnection(diskSource->GetOutputPort());
  vtkSmartPointer<vtkActor> diskActor =
    vtkSmartPointer<vtkActor>::New();
  diskActor->SetMapper(diskMapper);

  // Create a renderer, render window, and interactor
  vtkSmartPointer<vtkRenderer> renderer =
    vtkSmartPointer<vtkRenderer>::New();
  vtkSmartPointer<vtkRenderWindow> renderWindow =
    vtkSmartPointer<vtkRenderWindow>::New();
  renderWindow->AddRenderer(renderer);
  vtkSmartPointer<vtkRenderWindowInteractor> renderWindowInteractor =
    vtkSmartPointer<vtkRenderWindowInteractor>::New();
  renderWindowInteractor->SetRenderWindow(renderWindow);

  renderer->AddActor(edgeActor);
  renderer->AddActor(diskActor);
  renderer->SetBackground(.3, .6, .3); // Background color green

  renderWindow->Render();
  renderWindowInteractor->Start();
  
  return EXIT_SUCCESS;
}

File: src/CSharp/Meshes/BoundaryEdges.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Windows.Forms;
using System.Diagnostics;

using Kitware.VTK;

namespace ActiViz.Examples {
   public partial class Form1 : Form {
      public Form1() {
         InitializeComponent();
      }


      private void renderWindowControl1_Load(object sender, EventArgs e) {
         try {
            BoundaryEdges();
         }
         catch(Exception ex) {
            MessageBox.Show(ex.Message, "Exception", MessageBoxButtons.OK);
         }
      }


      private void BoundaryEdges() {
         vtkDiskSource diskSource = vtkDiskSource.New();
         diskSource.Update();

         vtkFeatureEdges featureEdges = vtkFeatureEdges.New();
#if VTK_MAJOR_VERSION_5
         featureEdges.SetInputConnection(diskSource.GetOutputPort());
#else
         featureEdges.SetInputData(diskSource);
#endif
         featureEdges.BoundaryEdgesOn();
         featureEdges.FeatureEdgesOff();
         featureEdges.ManifoldEdgesOff();
         featureEdges.NonManifoldEdgesOff();
         featureEdges.Update();

         // Visualize
         vtkPolyDataMapper edgeMapper = vtkPolyDataMapper.New();
#if VTK_MAJOR_VERSION_5
         edgeMapper.SetInputConnection(featureEdges.GetOutputPort());
#else
         edgeMapper.SetInputData(featureEdges);
#endif
         vtkActor edgeActor = vtkActor.New();
         edgeActor.GetProperty().SetLineWidth(3);
         edgeActor.SetMapper(edgeMapper);

         vtkPolyDataMapper diskMapper = vtkPolyDataMapper.New();
#if VTK_MAJOR_VERSION_5
         diskMapper.SetInputConnection(diskSource.GetOutputPort());
#else
         diskMapper.SetInputData(diskSource);
#endif
         vtkActor diskActor = vtkActor.New();
         diskActor.SetMapper(diskMapper);

         // get a reference to the renderwindow of our renderWindowControl1
         vtkRenderWindow renderWindow = renderWindowControl1.RenderWindow;
         // renderer
         vtkRenderer renderer = renderWindow.GetRenderers().GetFirstRenderer();
         // set background color
         renderer.SetBackground(0.3, 0.6, 0.3);
         // add our actor to the renderer
         renderer.AddActor(diskActor);
         renderer.AddActor(edgeActor);
      }
   }
}
